src/latexdocument.cpp

```diff
@@ -1412,6 +1440,35 @@ QMultiHash<QDocumentLineHandle *, int> LatexDocument::getLabels(const QString &n
 	return std::move(result);
 }
 
+QMultiHash<QDocumentLineHandle *, int> LatexDocument::getCommandDefinitions(const QString &name)
+{
+	QHash<QDocumentLineHandle *, int> result;
+	foreach (const LatexDocument *elem, getListOfDocs()) {
+		QMultiHash<QDocumentLineHandle *, ReferencePair>::const_iterator it;
+		for (it = elem->mUserCommandGoto.constBegin(); it != elem->mUserCommandGoto.constEnd(); ++it) {
+			ReferencePair rp = it.value();
+			if (rp.name == name && elem->indexOf(it.key()) >= 0) {
+				result.insert(it.key(), rp.start);
+			}
+		}
```

Comment:
Here we could instead compare `name` to the `CodeSnippet`s stored in `mUserCommandList`, in order to get rid of `mUserCommandGoto`.

Reply:
I would prefer usind mUserCommandList for all functionality. It might need extensions, but i am not a fan of redundant structures which are forgotten in the next refactorizatiom

Reply:
Okay, I will look into this. I think storing a struct containing both the name (for goto definition) and the CodeSnippet (for completion) would be a good idea then.

Reply:
This is part of the refactoring.